src/main/java/io/vertx/core/impl/VertxImpl.java

```diff
@@ -1126,9 +1126,6 @@ private ThreadFactory createThreadFactory(long maxExecuteTime, TimeUnit maxExecu
     return runnable -> {
       VertxThread thread = threadFactory.newVertxThread(runnable, prefix + threadCount.getAndIncrement(), worker, maxExecuteTime, maxExecuteTimeUnit);
       checker.registerThread(thread, thread);
-      // Vert.x threads are NOT daemons - we want them to prevent JVM exit so embedded user doesn't
```

Comment:
Or maybe we could just change the code here, to check the daemon option in VertxOptions, to reset the daemon flag of the thread. This could keep the old behavior if users do not change anything, and could also make the thread daemon if users set daemon to true in VertxOptions. WDYT? @vietj 

Thanks.